pkgs/applications/radio/gnuradio/pkgs/srcs.nix

```diff
@@ -0,0 +1,113 @@
+{ lib
+, ver
+# Needed for fetching sources
+, pkgs
```

Comment:
`pkgs` is only needed for the fetchers? Why not just handing over the fetchers?

Reply:
Yes. Well I could have put each fetcher used in the arguments, but then I would have had to put each fetcher in the arguments of `wrapper.nix` and also inherit each fetcher in the call when I `import ./pkgs/srcs.nix` (in `wrapper.nix`). I too don't like it that much and there's also only 1 fetcher used currently. My intention was that a new definition of a gnuradio package with a source from somewhere else then GitHub will require changing the arguments passed between the different files.

Perhaps it would be better indeed to put the sources inside the expressions themselves, where each fetcher is available because the expressions are evaluated with `callPackage`. What I wanted to avoid was having duplicate code in each gnuradio package that will define a package has only a certain source for 3.7/3.8. I will think about doing it as best as possible.

Reply:
Would it work to call `./pkgs/srcs.nix` with `callPackages` instead of `import`?
Otherwise I would go with your suggestion no 2, to put the sources inside the package derivations themselves. That is more compact and more readable?

Reply:
I agree that it's more compact and more readable. I pushed 6c20dd4f91ca76864caa647d19dcb13ba63a5293 now (which will be squashed eventually) that removes `srcs.nix` and there is not much duplicate code going on. `mkDerivation.nix` is where the shared logic of "Don't build this derivation for an incompatible gnuradio" is going on.